### ISSUE TITLE
[Improvement] Add warning when marking an object val as @SharedImmutable

### ISSUE THREAD

**Issue by benasher44** [JetBrains/kotlin-native#3049]
Take this example code:

```kotlin
object MySingleton {
    @SharedImmutable
    val foo: String
}
```

It would be nice to have have a warning that says that the variable is already shared and frozen (according to last few bullets here: https://github.com/JetBrains/kotlin-native/blob/master/CONCURRENCY.md). It'd be a nice opportunity to teach folks about the native concurrency, which is quite different at first. I have personally written code like that, and then I went back and cleaned it up, once I better understood the concurrency model.

**Comment by fluidsonic** [comment]
In addition an error should be raised if a property is marked with `@ThreadLocal` or `@SharedImmutable` which doesn't support it.

I've marked an object val as `@ThreadLocal` and assumed it will just work - but it doesn't.
You can even annotate class properties without warning or error.

**Comment by SvyatoslavScherbina** [comment]
Fixed in 1.4.
